File: app/src/main/java/com/liushu/crazyandroid/ui/main/fragment/FourFragment.java
package com.liushu.crazyandroid.ui.main.fragment;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ImageView;
import android.widget.TextView;

import com.jaydenxiao.common.base.BaseFragment;
import com.liushu.crazyandroid.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by liushu on 2017/1/26.
 */

public class FourFragment extends BaseFragment {


    @BindView(R.id.iv_back)
    ImageView mIvBack;
    @BindView(R.id.tv_title)
    TextView mTvTitle;
    @BindView(R.id.webView)
    WebView mWebView;
    private WebSettings settings;

    @Override
    protected int getLayoutResource() {
        return R.layout.fragment_four;
    }

    @Override
    public void initPresenter() {

    }

    @Override
    protected void initView() {
        mTvTitle.setText("版本号：" + getVersionName(mContext));
        settings = mWebView.getSettings();
        settings.setJavaScriptEnabled(true); //如果访问的页面中有Javascript，则mWebView必须设置支持Javascript
        settings.setJavaScriptCanOpenWindowsAutomatically(true);
        settings.setSupportZoom(true); //支持缩放
        settings.setBuiltInZoomControls(true); //支持手势缩放
        settings.setDisplayZoomControls(false); //是否显示缩放按钮

        // >= 19(SDK4.4)启动硬件加速，否则启动软件加速
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            mWebView.setLayerType(View.LAYER_TYPE_HARDWARE, null);
            settings.setLoadsImagesAutomatically(true); //支持自动加载图片
        } else {
            mWebView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
            settings.setLoadsImagesAutomatically(false);
        }

        settings.setUseWideViewPort(true); //将图片调整到适合mWebView的大小
        settings.setLoadWithOverviewMode(true); //自适应屏幕
        settings.setDomStorageEnabled(true);
        settings.setSaveFormData(true);
        settings.setSupportMultipleWindows(true);
        settings.setAppCacheEnabled(true);
        settings.setCacheMode(WebSettings.LOAD_DEFAULT); //优先使用缓存

        mWebView.setHorizontalScrollbarOverlay(true);
        mWebView.setHorizontalScrollBarEnabled(false);
        mWebView.setOverScrollMode(View.OVER_SCROLL_NEVER); // 取消mWebView中滚动或拖动到顶部、底部时的阴影
        mWebView.setScrollBarStyle(View.SCROLLBARS_INSIDE_OVERLAY); // 取消滚动条白边效果
        mWebView.requestFocus();

        mWebView.loadUrl("file:///android_asset/about.html");
        mWebView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                view.loadUrl(url);
                mIvBack.setVisibility(View.VISIBLE);
                return true;
            }
        });
    }

    // 获取当前应用的版本号
    public static String getVersionName(Context context) {
        try {
            PackageManager packageManager = context.getPackageManager();
            PackageInfo packInfo = packageManager.getPackageInfo(context.getPackageName(), 0);
            String version = packInfo.versionName;
            if (!TextUtils.isEmpty(version)) {
                return version;
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "";
    }

    @OnClick(R.id.iv_back)
    public void onViewClicked() {
        if (mWebView.canGoBack()) {
            mWebView.goBack();//返回上一页面

            if (!mWebView.canGoBack()) {
                mIvBack.setVisibility(View.INVISIBLE);
            }
        }
    }
}

File: app/src/main/java/com/liushu/crazyandroid/ui/main/fragment/FourFragment.kt
package com.liushu.crazyandroid.ui.main.fragment

import android.content.Context
import android.os.Build
import android.text.TextUtils
import android.view.View
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import com.jaydenxiao.common.base.BaseFragment
import com.liushu.crazyandroid.R
import kotlinx.android.synthetic.main.fragment_four.*

/**
 * Created by liushu on 2017/1/26.
 */

class FourFragment : BaseFragment() {

    private var settings: WebSettings? = null

    override fun getLayoutResource(): Int {
        return R.layout.fragment_four
    }

    override fun initPresenter() {

    }

    override fun initView() {
        tv_title!!.text = "版本号：${getVersionName(mContext)}"
        settings = webView!!.settings
        settings!!.javaScriptEnabled = true //如果访问的页面中有Javascript，则mWebView必须设置支持Javascript
        settings!!.javaScriptCanOpenWindowsAutomatically = true
        settings!!.setSupportZoom(true) //支持缩放
        settings!!.builtInZoomControls = true //支持手势缩放
        settings!!.displayZoomControls = false //是否显示缩放按钮

        // >= 19(SDK4.4)启动硬件加速，否则启动软件加速
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            webView!!.setLayerType(View.LAYER_TYPE_HARDWARE, null)
            settings!!.loadsImagesAutomatically = true //支持自动加载图片
        } else {
            webView!!.setLayerType(View.LAYER_TYPE_SOFTWARE, null)
            settings!!.loadsImagesAutomatically = false
        }

        settings!!.useWideViewPort = true //将图片调整到适合mWebView的大小
        settings!!.loadWithOverviewMode = true //自适应屏幕
        settings!!.domStorageEnabled = true
        settings!!.saveFormData = true
        settings!!.setSupportMultipleWindows(true)
        settings!!.setAppCacheEnabled(true)
        settings!!.cacheMode = WebSettings.LOAD_DEFAULT //优先使用缓存

        webView!!.setHorizontalScrollbarOverlay(true)
        webView!!.isHorizontalScrollBarEnabled = false
        webView!!.overScrollMode = View.OVER_SCROLL_NEVER // 取消mWebView中滚动或拖动到顶部、底部时的阴影
        webView!!.scrollBarStyle = View.SCROLLBARS_INSIDE_OVERLAY // 取消滚动条白边效果
        webView!!.requestFocus()

        webView!!.loadUrl("file:///android_asset/about.html")
        webView!!.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                view.loadUrl(url)
                iv_back!!.visibility = View.VISIBLE
                return true
            }
        }
    }

 /*   @OnClick(R.id.iv_back)
    fun onViewClicked() {
        if (webView!!.canGoBack()) {
            webView!!.goBack()//返回上一页面

            if (!webView!!.canGoBack()) {
                iv_back!!.visibility = View.INVISIBLE
            }
        }
    }*/

    companion object {

        // 获取当前应用的版本号
        fun getVersionName(context: Context): String {
            try {
                val packageManager = context.packageManager
                val packInfo = packageManager.getPackageInfo(context.packageName, 0)
                val version = packInfo.versionName
                if (!TextUtils.isEmpty(version)) {
                    return version
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return ""
        }
    }
}
